main.cpp

```diff
@@ -547,7 +547,7 @@ bool llama_eval(
 
     const int d_key = n_embd/n_head;
 
-    static size_t buf_size = 512u*1024*1024;
+    static size_t buf_size = hparams.n_ctx*1024*1024;
```

Comment:
This is possibly correct change, but I need to verify

```suggestion
     // TODO: check if this size scales with n_ctx linearly and remove constant. somehow I feel it wasn't the case
    // static size_t buf_size = hparams.n_ctx*1024*1024;
    static size_t buf_size = 512u*1024*1024;
```